packages/paste-core/layout/grid/src/Grid.tsx

```diff
@@ -47,12 +51,14 @@ const Grid = React.forwardRef<HTMLDivElement, GridProps>(
               })
             : child
         ),
-      [children]
+      [children, vertical, equalColumnHeights, gutter]
     );
 
-    const gutterStyles = React.useMemo(() => getGutterStyles(gutter), [gutter]);
+    const gutterStyles = React.useMemo(() => getGutterStyles(gutter, vertical), [gutter, vertical]);
     const flexDirection = React.useMemo(() => getFlexDirection(vertical), [vertical]);
 
+    const gridStyles = {...gutterStyles, marginTop, marginBottom};
```

Comment:
nit: can we undo this as it adds some unnecessary overhead to create a new object

Reply:
Sure thing 👍